spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -48,9 +50,12 @@ public SourceConfigurer from(Class<?> app) {
 	}
 
 	public void run(String[] parentArgs) {
-		ConfigurableApplicationContext parentContext = AggregateApplication.createParentContext(parentArgs);
-		SharedChannelRegistry sharedChannelRegistry = parentContext.getBean(SharedChannelRegistry.class);
-		List<AppConfigurer> apps = new ArrayList<AppConfigurer>();
+		ConfigurableApplicationContext parentContext = this.parentContext != null
+				? this.parentContext
+				: AggregateApplication.createParentContext(parentArgs);
+		SharedChannelRegistry sharedChannelRegistry = parentContext
```

Comment:
If there's not `SharedChannelRegistry`, then channels will not be shared, I believe. It shouldn't be fundamentally hard to do a `registerSingleton` in the parent though.